packages/remix-server-runtime/responses.ts

```diff
@@ -13,11 +15,15 @@ export type TypedDeferredData<Data extends Record<string, unknown>> = Pick<
   data: Data;
 };
 
+// TODO: The RR version of this is the same minus the generic.  Should we add
+// that over there and remove this?
 export type DeferFunction = <Data extends Record<string, unknown>>(
   data: Data,
   init?: number | ResponseInit
 ) => TypedDeferredData<Data>;
 
+// TODO: The RR version of this is identical except it's generic doesn't
+// `extends unknown`.  Should we add that over there and remove this?
 export type JsonFunction = <Data extends unknown>(
```

Comment:
Unsure if these generics make sense in RR if we're moving to an [`element: (data) => <p>{data}</p>`](https://github.com/orgs/remix-run/projects/5?pane=issue&itemId=18313356) type API in the future for better typing

Reply:
Although it ca also be inferred, it still makes sense to define the actual type that's passed in `json` imo

Reply:
Going to split the generic out for a v2 decision - not needed in 1.15.0